gui/src/components/form/SubmitButton.svelte

```diff
@@ -0,0 +1,17 @@
+<script lang="ts">
+  import Button from '../Button.svelte';
+
+  // This is a right-aligned submit button for forms.
+  // Used for create/edit forms where the code below
+  // would otherwise be highly repetitive.
+</script>
+
+<div><Button type="submit"><slot /></Button></div>
+
+<style>
+  div {
+    display: flex;
+    justify-content: flex-end;
```

Comment:
Why is this part of the button? Seems like the layout logic should be part of the form/container, such as a slot for form buttons. Consider what happens if you want _two_ buttons. How would that work? Generally, it's better to have a little bit more verbosity than it is to bake policy in too deeply into the component hierarchy.

Reply:
As it is now, this is supposed to be like a "block" component for forms that can be used and replaced with ease.

The original approach, when this `div` was called `.buttons`, was to support multiple buttons but I don't think it's worthwhile setting that up until we actually have a need for it.

It's likely that I'll refactor this in the next week with **UPDATE** forms for components, where we probably want both a primary "save changes" and secondary "cancel" and/or even "delete" buttons.

Anyways, this PR is intended to clean up the current state of things and tie up loose ends before adding full CRUD GUI which is one of the handful of priorities I have right now.